api/src/main/java/net/md_5/bungee/api/event/ServerChatEvent.java

```diff
@@ -0,0 +1,57 @@
+package net.md_5.bungee.api.event;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.Connection;
+import net.md_5.bungee.api.plugin.Cancellable;
+import net.md_5.bungee.chat.ComponentSerializer;
+
+import java.util.function.Supplier;
+
+import static lombok.AccessLevel.NONE;
+
+/**
+ * Event called when a server sends a message to a player.
+ */
+@Data
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+public class ServerChatEvent extends TargetedEvent implements Cancellable
+{
+
+    /**
+     * Cancelled state.
+     */
+    private boolean cancelled;
+    /**
+     * Text contained in this chat.
+     */
+    private BaseComponent[] message = null;
+    @Getter(NONE)
+    @Setter(NONE)
+    private Supplier<BaseComponent[]> messageSupplier;
+    /**
+     * Type of message/display location
+     */
+    private ChatMessageType position;
+
+    public ServerChatEvent(Connection sender, Connection receiver, String json, byte position)
+    {
+        super( sender, receiver );
+        this.messageSupplier = () -> ComponentSerializer.parse( json );
```

Comment:
Uh Bungee isn't Java 8. How did you even get this to compile :\

Reply:
:man_shrugging:
I'll convert it to the verbose version

Reply:
Ah, this is how

```diff
diff --git a/api/pom.xml b/api/pom.xml
index 5432a6eb..025e1498 100644
--- a/api/pom.xml
+++ b/api/pom.xml
@@ -2,6 +2,18 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
+                    <target>8</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 
     <parent>
         <groupId>net.md-5</groupId>
```